src/AppBundle/Controller/EntityController.php

```diff
@@ -81,41 +81,57 @@ private function setProgramAndEvent(): void
     }
 
     /**
-     * Check the request and if the programTitle is set, find and set
-     * $this->program with the corresponding entity.
+     * Check the request and if the programId is set, find and set $this->program with the corresponding entity.
      * @throws NotFoundHttpException
      */
     private function setProgram(): void
     {
-        $programTitle = $this->request->get('programTitle');
-        if ($programTitle) {
-            $this->program = $this->em->getRepository(Program::class)
-                ->findOneBy(['title' => $programTitle]);
-
-            if (!is_a($this->program, Program::class)) {
-                throw new NotFoundHttpException('error-not-found');
-            }
+        $programId = $this->request->get('programId');
+
+        if (!$programId) {
+            return;
+        }
+
+        $repo = $this->em->getRepository(Program::class);
+
+        if (is_numeric($programId)) {
```

Comment:
Do you want to test for an integer? `is_numeric` will give `True` for floats.

Reply:
Perfect! I was right and wrong at the same time. Pretty typical.

That function does look like what we need. It even throws false if handed an integer.